core/dbt/graph/queue.py

```diff
@@ -25,8 +25,14 @@ class GraphQueue:
     the same time, as there is an unlocked race!
     """
 
-    def __init__(self, graph: nx.DiGraph, manifest: Manifest, selected: Set[UniqueId]) -> None:
-        self.graph = graph
+    def __init__(
+        self,
+        graph: nx.DiGraph,
+        manifest: Manifest,
+        selected: Set[UniqueId],
+        preserve_edges: bool = True,
+    ) -> None:
+        self.graph = graph if preserve_edges else nx.classes.function.create_empty_copy(graph)
```

Comment:
Could you add a comment above the "create_empty_copy" saying something like "'create_empty_copy' will remove all of the edges in the graph"?  Because to me, "create_empty_copy" was not intuitive and I had to look it up.

Reply:
Good call! Will do 👍